ios/engine/KMEI/KeymanEngine/Classes/Migrations.swift

```diff
@@ -236,60 +205,59 @@ public enum Migrations {
       let resources = possibleHistories[possibleHistories.count-1].resources
 
       resources.forEach { res in
-        if let kbd = res as? InstallableKeyboard {
-          var userKeyboards = Storage.active.userDefaults.userKeyboards
-
+        if let kbd = res as? FullKeyboardID {
           // Does not remove the deprecated keyboard's files - just the registration.
-          userKeyboards?.removeAll(where: { kbd2 in
-            return kbd.id == kbd2.id && kbd.languageID == kbd2.languageID
-          })
-          Storage.active.userDefaults.userKeyboards = userKeyboards
-        } else if let lex = res as? InstallableLexicalModel {
-          var userModels = Storage.active.userDefaults.userLexicalModels
-
+          if let match = userKeyboards.first(where: { $0.fullID == kbd }) {
+            if match.fullID == Defaults.keyboard.fullID {
+              currentDefaultKbdVersion = Version(match.version)
+            }
+            oldDefaultKbd = match
+          } else {
+            hasVersionDefaults = false
+          }
+        } else if let lex = res as? FullLexicalModelID {
           // Parallels the issue with deprecated files for keyboards.
-          userModels?.removeAll(where: { lex2 in
-            return lex.id == lex2.id && lex.languageID == lex2.languageID
-          })
-          Storage.active.userDefaults.userLexicalModels = userModels
+          if let match = userModels.first(where: { $0.fullID == lex }) {
+            if match.fullID == Defaults.lexicalModel.fullID {
+              currentDefaultLexVersion = Version(match.version)
+            }
+            oldDefaultLex = match
+          } else {
+            hasVersionDefaults = false
+          }
         } // else Not yet implemented
       }
+
+      // The user has customized their default resources; Keyman will refrain from
+      // changing the user's customizations.
+      if !hasVersionDefaults {
```

Comment:
Could this be refactored to:

```suggestion
      if !oldDefaultLex || !oldDefaultKbd {
```

That would allow us to eliminate that additional variable.

Reply:
If a user is (somehow) updating from KMW 10, there was no default lexical model.  This is why the variable is necessary; `hasVersionDefaults` only checks for lexical models if that version had a default for it.